src/equations/shallow_water_2d.jl

```diff
@@ -50,14 +56,20 @@ References for the SWE are many but a good introduction is available in Chapter
 struct ShallowWaterEquations2D{RealT<:Real} <: AbstractShallowWaterEquations{2, 4}
   gravity::RealT # gravitational constant
   H0::RealT      # constant "lake-at-rest" total water height
+  threshold_limiter::RealT  # Threshold to use in PositivityPreservingLimiterShallowWater on water height,
+                             # as a (small) shift on the initial condition and cutoff before the
+                             # next time step.
+   threshold_wet::RealT      # Threshold to be applied on water height to define when the flow is "wet"
+                             # before calculating the numerical flux.
 end
 
 # Allow for flexibility to set the gravitational constant within an elixir depending on the
 # application where `gravity_constant=1.0` or `gravity_constant=9.81` are common values.
-# The reference total water height H0 defaults to 0.0 but is used for the "lake-at-rest"
-# well-balancedness test cases
-function ShallowWaterEquations2D(; gravity_constant, H0=0.0)
-  ShallowWaterEquations2D(gravity_constant, H0)
+# The reference total water height H0 is an artefact from the old calculation of the lake_at_rest_error
+# Strict default values for thresholds that performed great in several numerical experiments
+function ShallowWaterEquations2D(; gravity_constant, H0=0.0,
+                                 threshold_limiter=1e-13, threshold_wet=1e-15)
+  ShallowWaterEquations2D(gravity_constant, H0, threshold_limiter, threshold_wet)
 end
```

Comment:
Yes.

Reply:
I am not completely familiar with all data types in Julia. But, I changed the data type of both limiters in the SWE struct from `RealT` to `Float64`. In that case, your example worked on my local machine. Is this an edit, I can consider?

Reply:
No. Let's ask it this way. If you were using `Float32` instead of `Float64`, which default values of the parameters would you choose? Still something like `10 * eps() ≈ 1.0e-15` and `1000 * eps() ≈ 1.0e-13`?

Reply:
Also, the comment regarding `H0` should be removed / updated. It is still true that we use the reference water height to compute the lake-at-rest error.